RELEASES.md

```diff
@@ -19,47 +38,50 @@
 
-* For minor releases create a new release branch and
-  * update versions in release branch, e.g. [#2803](https://github.com/elastic/apm-server/pull/2803/files)
-  * update versions in `major.x` branch to next minor version, e.g. [#2804](https://github.com/elastic/apm-server/pull/2804)
-
-* Update to latest changes of [beats](https://github.com/elastic/beats/pulls/)
-  * Update `BEATS_VERSION` to the release version in the top-level Makefile
-  * When beats has merged all PRs and for minor releases created the new branch, run `make update-beats` and commit the changes.
-
 * Ensure a branch or tag is created for the [go-elasticsearch](https://github.com/elastic/go-elasticsearch) library and update to it.
 
   `go get github.com/elastic/go-elasticsearch/v$major@$major.$minor`
 
-* The following may also need to be updated manually:
-    * APM Overview's [release highlights](https://github.com/elastic/apm-server/blob/main/docs/guide/apm-release-notes.asciidoc) - Anything exciting across the APM stack!
-    * APM Overview's [breaking changes](https://github.com/elastic/apm-server/blob/main/docs/guide/apm-breaking-changes.asciidoc) - Any breaking changes across the APM stack.
-    * APM Server's [breaking changes](https://github.com/elastic/apm-server/blob/main/docs/breaking-changes.asciidoc) - Any APM Server breaking changes.
-    * APM Server's [upgrade guide](https://github.com/elastic/apm-server/blob/main/docs/upgrading.asciidoc).
+* Collaborate with the docs team on any release highlights or breaking changes that should be included in the APM Server guide. 
+
+* Update [.mergify.yml](https://github.com/elastic/apm-server/blob/main/.mergify.yml) with a new backport rule for the next version.
 
-* For major releases, update and smoke test the dev quick start [`docker-compose.yml`](https://github.com/elastic/apm-server/blob/main/docs/guide/docker-compose.yml).
+* Create a test plan
 
-## After feature freeze
+  Create a github issue for testing the release branch, e.g. [8.4.0 test plan](https://github.com/elastic/apm-server/issues/8705)
+  It should contain 
+  * a link to all PRs in the APM Server repository that need to be tested manually. Use the test-plan* labels and the version labels 
+    to create an overview over the PRs that need testing. For example, [test plan link for 8.3.0](https://github.com/elastic/apm-server/issues?q=label%3Atest-plan+is%3Aclosed+label%3Av8.3.0).
+  * add test scenarios outside the APM Server repository that require manual testing, such as test scenarios on ESS, that are not covered by automated tests or 
+    OS compatibility smoke tests if support for new operating systems is introduced.  
 
-* Update [.mergify.yml](https://github.com/elastic/apm-server/blob/main/.mergify.yml) with a new backport rule for the next version.
+* Test the release branch
+
+  * When the first build candidate (BC) is published, you can start testing.
```

Comment:
Maybe explicitly state here that we should always test with build candidates, to verify that we're testing the artifacts that will be distributed.